control/src/lib.rs

```diff
@@ -495,7 +488,7 @@ impl<T: Config> Pallet<T> {
 		T::Currency::transfer(
 			T::ProtocolTokenId::get(),
 			&creator,
-			&T::GameDAOTreasury::get(),
+			&treasury,
 			T::CreationFee::get(),
```

Comment:
I'd change the naming. In this context, it's not a creation fee anymore but some initial deposit.